imagersite/imager_profile/tests/test_views.py

```diff
@@ -0,0 +1,52 @@
+from django.contrib.auth.models import User, AnonymousUser
+from django.test import TestCase, RequestFactory
+from ..views import profile_view
+from ..models import ImagerProfile
+from model_mommy import mommy
+from django.urls import reverse_lazy
+from django.http import Http404
+
+
+class ProfileViewTest(TestCase):
+    def setUp(self):
+        self.factory = RequestFactory()
+        for _ in range(2):
+            user = mommy.make(User)
+            mommy.make(ImagerProfile, user=user)
+
+    def test_user_views_own_profile(self):
+        """Test that user's own profile renders."""
+        request = self.factory.get(reverse_lazy('profile'))
+        request.user = User.objects.first()
+        response = profile_view(request)
+
+        self.assertTrue(response.status_code == 200)
+
+    def test_user_views_another_profile(self):
+        """Test user can render another user's profile."""
+        user = User.objects.first()
+        request = self.factory.get(reverse_lazy('named_profile', args=[
+            User.objects.exclude(username=user.username).first().username]))
+        request.user = User.objects.first()
+        response = profile_view(request)
```

Comment:
This does not test passing in a username in the url. The only part that matters for url parameters is what arguments are passed into the view function itself if you are doing a unit test like this.

Reply:
If you use the client to go to this route, then what you have done in the `reverse_lazy` will apply.